# Multiple Linear Regression

Adjusted R-squared:

(1) The R-squared measures how much of the total variability is explained by our model

(2) Multiple regressions are always better than simple ones, as with each addtional variable you add, the explanatory power may only increase or stay the same.

(3) Adjusted R-squared is always less than R-squared. The adjusted R-squared penalize excessive use of variables.


# Import the relevant libraries

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import seaborn as sns
sns.set()

# Load data

In [3]:
data = pd.read_csv('1.02. Multiple linear regression.csv')

In [7]:
data.describe()

SAT        GPA  Rand 1,2,3
count    84.000000  84.000000   84.000000
mean   1845.273810   3.330238    2.059524
std     104.530661   0.271617    0.855192
min    1634.000000   2.400000    1.000000
25%    1772.000000   3.190000    1.000000
50%    1846.000000   3.380000    2.000000
75%    1934.000000   3.502500    3.000000
max    2050.000000   3.810000    3.000000

In [8]:
y = data['GPA']
x1 = data[['SAT', 'Rand 1,2,3']]

In [11]:
x = sm.add_constant(x1.values)
results = sm.OLS(y,x).fit()

In [12]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    GPA   R-squared:                       0.407
Model:                            OLS   Adj. R-squared:                  0.392
Method:                 Least Squares   F-statistic:                     27.76
Date:                Sat, 25 Jan 2020   Prob (F-statistic):           6.58e-10
Time:                        23:59:45   Log-Likelihood:                 12.720
No. Observations:                  84   AIC:                            -19.44
Df Residuals:                      81   BIC:                            -12.15
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.2960      0.417      0.710      0.480      -0.533       1.125
x1             0.0017      0.000      7.432      0.000       0.001       0.002
x2            -0.0083      0.027     -0.304      0.762      -0.062       0.046
==============================================================================
Omnibus:                       12.992   Durbin-Watson:                   0.948
Prob(Omnibus):                  0.002   Jarque-Bera (JB):               16.364
Skew:                          -0.731   Prob(JB):                     0.000280
Kurtosis:                       4.594   Cond. No.                     3.33e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.33e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [13]:
y = data['GPA']
x1 = data['SAT']

In [14]:
x = sm.add_constant(x1.values)
results = sm.OLS(y,x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    GPA   R-squared:                       0.406
Model:                            OLS   Adj. R-squared:                  0.399
Method:                 Least Squares   F-statistic:                     56.05
Date:                Sun, 26 Jan 2020   Prob (F-statistic):           7.20e-11
Time:                        11:23:12   Log-Likelihood:                 12.672
No. Observations:                  84   AIC:                            -21.34
Df Residuals:                      82   BIC:                            -16.48
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.2750      0.409      0.673      0.503      -0.538       1.088
x1             0.0017      0.000      7.487      0.000       0.001       0.002
==============================================================================
Omnibus:                       12.839   Durbin-Watson:                   0.950
Prob(Omnibus):                  0.002   Jarque-Bera (JB):               16.155
Skew:                          -0.722   Prob(JB):                     0.000310
Kurtosis:                       4.590   Cond. No.                     3.29e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.29e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Based on above, our regression model is <strong>GPA = 0.2750 + 0.0017 * SAT</strong>

## F-statistic


The F-statistic is used for testing the overall signigicance of the model.

F-test: <br />


### Regression assumptions:
(1) Linearity <br />
(2) No endogeneity - e.g., omitted variables <br />
(3) Normality and homoscedasticity <br />
(4) No autocorrelation <br />
(5) No multicollinearity <br />


open mx_dummy_variables.jpynb  for more prediction